### PR TITLE
Add adapters for go-kit metrics and a factory for expvar

### PR DESCRIPTION
**Pull request by yurishkuro**
This implements the pattern I was asking for in https://github.com/go-kit/kit/issues/378, simplified for our needs.

### CODE REVIEW

File: metrics/go-kit/expvar/factory.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+type factory struct {
+	scope   string
+	buckets int
+}
+
+func (f factory) subScope(name string) string {
+	return f.scope + "_" + name
+}
+
+func (f factory) Counter(name string, tags map[string]string) metrics.Counter {
```
Comment (black-adder) [inline metrics/go-kit/expvar/factory.go]: does go-kit metrics not have a way to deal with tags?

File: metrics/go-kit/expvar/factory.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+func (f factory) Timer(name string, tags map[string]string) metrics.Timer {
+	return xkit.NewTimer(kit.NewHistogram(f.subScope(name), f.buckets))
+}
+
+func (f factory) Gauge(name string, tags map[string]string) metrics.Gauge {
+	return xkit.NewGauge(kit.NewGauge(f.subScope(name)))
+}
+
+// Namespace is a no-op for expvar, since i
```
Comment (black-adder) [inline metrics/go-kit/expvar/factory.go]: eh?

File: glide.yaml [hunk 3]
```diff
@@ -1,3 +1,7 @@
 package: github.com/uber/jaeger-lib
 import:
 - package: github.com/codahale/hdrhistogram
+- package: github.com/go-kit/kit
+  subpackages:
```
Comment (yurishkuro) [inline glide.yaml]: sorry, I tried to add them manually, but glide complained. Removed.

File: metrics/go-kit/expvar/factory.go (hunk 1 above)
Comment (yurishkuro) [inline metrics/go-kit/expvar/factory.go]: oops, you're correct, I got focused on expvar implementation that does not support tags

**Pull request merged**